protobuf/types/serialize.go

```diff
@@ -0,0 +1,72 @@
+package types
+
+import (
+	"sort"
+	"strconv"
+
+	"github.com/mesg-foundation/engine/hash/hashserializer"
+)
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Struct) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.Add("1", mapValue(data.Fields))
+	return ser.HashSerialize()
+}
+
+type mapValue map[string]*Value
+
+// HashSerialize returns the hashserialized string of this type
+func (data mapValue) HashSerialize() string {
+	ser := hashserializer.New()
+	keys := make([]string, 0, len(data))
+	for k := range data {
+		keys = append(keys, k)
+	}
+	sort.Strings(keys)
+	for _, key := range keys {
+		ser.Add(key, data[key])
+	}
+	return ser.HashSerialize()
+}
+
+// HashSerialize returns the hashserialized string of this type
+func (data *Value) HashSerialize() string {
+	if data == nil {
+		return ""
+	}
+	ser := hashserializer.New()
+	ser.AddFloat("2", data.GetNumberValue())
```

Comment:
same comment here about the null value, let's add it even we know for now it will be discarded.